### ISSUE TITLE
Pin based authentication in systemd service

### ISSUE THREAD

**Issue by bdashore3** [egg-mode-rs/egg-mode#106]
Hi, I'm working with my twitter bot and created a systemd service for it. However, I need user authentication before I can proceed and I am trying to use pin-based auth.

The issue here is that when running in a systemd service, I can't provide the pin through the interactive terminal due to systemd not having an interactive shell

Is there any way to work with pin-based auth with systemd?

**Comment by QuietMisdreavus** [comment]
One thing you could try is to make it so you can run it interactively *once*, then save the tokens that come out so that the systemd service can still use them later. The examples in this repo all use this pattern - check out `examples/common/mod.rs` for that implementation.

**Comment by bdashore3** [comment]
So, I actually figured out another way to do this.

Even if I reuse the information I received from twitter, I still have to relogin with the pin and get new credentials every time I start the bot.

Systemd allows me to provide input/output using a TTY instance, so I linked a systemd process to a TTY shell where I inputted the pin and can start the bot.

Here's [the service itself](https://github.com/bdashore3/Simple-Retweets/blob/master/systemd/simplerts.service) if you're interested in taking a look
